main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -938,6 +943,37 @@ object Simplifier {
       case p => throw InternalCompilerException(s"Unsupported pattern '$p'.", xs.head.loc)
     }
 
+  private def extensibleMatch(label: Name.Label, exp: SimplifiedAst.Expr, sym1: Symbol.VarSym, exp1: SimplifiedAst.Expr, sym2: Symbol.VarSym, exp2: SimplifiedAst.Expr, tpe: MonoType, eff: Purity, loc: SourceLocation)(implicit flix: Flix): SimplifiedAst.Expr = {
+    import SimplifiedAst.*
+    //
+    // exp match {
+    //     case Label(sym1) => exp1
+    //     case sym2 => exp2
+    //
+    // Becomes
+    //
+    // let ext: tagType = exp;
+    // if (ext is Label) {
+    //     let sym1 = untag 0 ext;
+    //     exp1
+    // } else {
+    //     let sym2 = ext;
+    //     exp2
+    // }
+    //
+    val tagType = exp.tpe
+    val extName = Symbol.freshVarSym("ext", BoundBy.Let, exp.loc)(Scope.Top, flix)
+    val extVar = Expr.Var(extName, tagType, exp.loc)
+    val termTypes = MonoType.findExtensibleTermTypes(label, tagType)
+    val idx = 0 // Currently all extensible variants have 1 term.
```

Comment:
Is this true? I guess you will soon find out.

Reply:
That's how you wrote it in ConstraintGen, afaik